checker/src/org/checkerframework/checker/index/samelen/SameLenTransfer.java

```diff
@@ -149,27 +151,38 @@ private boolean isArrayLengthAccess(Node node) {
      * and b have SameLen of each other in the store.
      */
     private void refineEq(Node left, Node right, CFStore store) {
-        Receiver leftRec = FlowExpressions.internalReprOf(analysis.getTypeFactory(), left);
-        Receiver rightRec = FlowExpressions.internalReprOf(analysis.getTypeFactory(), right);
+        List<Receiver> receivers = new ArrayList<>();
+        List<AnnotationMirror> annos = new ArrayList<>();
 
-        AnnotationMirror leftAnno = getAnno(left);
-        AnnotationMirror rightAnno = getAnno(right);
-        AnnotationMirror combinedSameLen =
-                aTypeFactory.createCombinedSameLen(leftRec, rightRec, leftAnno, rightAnno);
+        for (Node internal : splitAssignments(left)) {
+            receivers.add(FlowExpressions.internalReprOf(analysis.getTypeFactory(), internal));
+            annos.add(getAnno(internal));
+        }
+
+        for (Node internal : splitAssignments(right)) {
+            receivers.add(FlowExpressions.internalReprOf(analysis.getTypeFactory(), internal));
+            annos.add(getAnno(internal));
+        }
+
+        AnnotationMirror combinedSameLen = aTypeFactory.createCombinedSameLen(receivers, annos);
 
         propagateCombinedSameLen(combinedSameLen, left, store);
     }
 
     /** Return n's annotation from the SameLen hierarchy. */
     AnnotationMirror getAnno(Node n) {
-        CFValue cfValue = analysis.getValue(n);
-        if (cfValue == null) {
+        if (n.isLValue()) {
```

Comment:
`analysis.getValue` fails if called on an lvalue. However, I need the annotations from lvalues when they are arrays in assignments. Consider this code:
`if ( (a = b) == c) {...}`, where `a`, `b`, and `c` are all arrays, and `a` has type `@SameLen("d")`. Afterwards, all three should have the type `@SameLen({"a", "b", "c", "d"})`, but in order to accomplish this we need to get the type of `a`, which is an lvalue.